contracts/peg/BasePSM.sol

```diff
@@ -0,0 +1,337 @@
+    /// First get oracle price of token
+    /// Then figure out how many dollars that amount in is worth by multiplying price * amount.
+    /// ensure decimals are normalized if on underlying they are not 18
+    function getMintAmountOut(uint256 amountIn)
+        public
+        view
+        override
+        returns (uint256 amountVoltOut)
+    {
+        Decimal.D256 memory oraclePrice = readOracle();
+        _validatePriceRange(oraclePrice);
+
+        uint256 voltPrice = oraclePrice.asUint256();
+
+        amountVoltOut = amountIn / voltPrice;
+    }
+
+    /// @notice calculate the amount of underlying out for a given `amountVoltIn` of VOLT
+    /// First get oracle price of token
+    /// Then figure out how many dollars that amount in is worth by multiplying price * amount.
+    /// ensure decimals are normalized if on underlying they are not 18
+    function getRedeemAmountOut(uint256 amountVoltIn)
+        public
+        view
+        override
+        returns (uint256 amountTokenOut)
+    {
+        Decimal.D256 memory oraclePrice = readOracle();
+        _validatePriceRange(oraclePrice);
+
+        uint256 voltPrice = oraclePrice.asUint256();
+
+        amountTokenOut = amountVoltIn * voltPrice;
+    }
+
+    /// @notice the maximum mint amount out
+    function getMaxMintAmountOut() external view override returns (uint256) {
+        return volt().balanceOf(address(this));
+    }
+
+    /// @notice the maximum redeem amount out
+    function getMaxRedeemAmountOut() external view returns (uint256) {
+        return underlyingToken.balanceOf(address(this));
+    }
+
+    /// @notice a flag for whether the current balance is above (true) or below (false) the reservesThreshold
+    function hasSurplus() external view override returns (bool) {
+        return balance() > reservesThreshold;
+    }
+
+    /// @notice an integer representing the positive surplus or negative deficit of contract balance vs reservesThreshold
+    function reservesSurplus() public view override returns (int256) {
+        return balance().toInt256() - reservesThreshold.toInt256();
+    }
+
+    /// @notice function from PCVDeposit that must be overriden
+    function balance() public view virtual override returns (uint256) {
+        return underlyingToken.balanceOf(address(this));
+    }
+
+    /// @notice returns address of token this contracts balance is reported in
+    function balanceReportedIn() public view override returns (address) {
+        return address(underlyingToken);
+    }
+
+    /// @notice override default behavior of not checking VOLT balance
+    function resistantBalanceAndVolt()
+        public
+        view
+        override
+        returns (uint256, uint256)
+    {
+        return (balance(), voltBalance());
+    }
+
+    // ----------- Internal Methods -----------
+
+    /// @notice Allocates a portion of escrowed PCV to a target PCV deposit
+    function _allocate(uint256 amount) internal virtual {
```

Comment:
we can remove

Reply:
the deposit function relies on this function, and we need the deposit function to adhere to the `IPCVDeposit` interface, would it make sense to leave the deposit function unimplemented in the base contract, but just have it present to adhere to the interface?

Reply:
we can remove the allocate function and make the deposit function have no logic

Reply:
your suggestion works and let's go with that, remove the deposit function in the basePSM contract and let the contract that inherits it decide what to do with it